src/main/java/net/minecraftforge/client/gui/ForgeIngameGui.java

```diff
@@ -570,7 +571,24 @@ public void renderFood(int width, int height, PoseStack mStack)
         boolean unused = false;// Unused flag in vanilla, seems to be part of a 'fade out' mechanic
 
         FoodData stats = minecraft.player.getFoodData();
-        int level = stats.getFoodLevel();
+        // Account for changes in max hunger here
+        int realHunger = stats.getFoodLevel();
+        int level;
+        // Only set level to 0 if the real hunger value is 0
+        if (realHunger == 0)
+        {
+            level = 0;
+        }
+        else
+        {
+            // Return a scaled value so that the HUD can still use the same logic as if the max was 20.
+            float scale = 20.0F / ForgeEventFactory.getMaxHunger(minecraft.player);
```

Comment:
I can buy that. I guess it comes down to if Forge wants to be opinionated on this. I could maybe see an issue where leaving it to the mod devs who change the max hunger could cause a potential incompatibility if multiple mods do this in one instance. Not sure what the precedent is for that, so I'll leave that as an open question for someone else to answer.

Reply:
Even if multiple mods were to be drawing additional drumstick lines, technically they would just be drawing them over eachother with the same data (thus not really conflicting beyond wasting time).  It would be simple to sort out at the modpack level with configuration files.